src/com/tekion/cricket/Match.java

```diff
@@ -0,0 +1,267 @@
+package com.tekion.cricket;
+
+import java.util.Scanner;
+import java.util.concurrent.ThreadLocalRandom;
+
+public class Match {
+    private final Team team1;
+    private final Team team2;
+    enum Winner{
+        TEAM1,
+        TEAM2,
+        TIE,
+        STARTED,
+        DRAW,
+    }
+    private Winner winner;
+    private final int totalAvailableBalls;
+
+    public Match()
```

Comment:
during re-orchestration of design principles, we may have to re-work on this portion.